woocommerce/woocommerce-android
# [Dynamic Dashboard] Coupons Card, part 1

Part of: #11508 


### Description
This PR is part of the work to add the coupons card, it adds the following:
1. It builds on top of https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3012 to add the needed logic to the repository.
2. It adds the ViewModel for the card with all the business logic.
3. It adds the logic of the range selection.
4. It adds a basic UI to test the logic.
5. Renaming and moving few classes around for a better organization.

The final UI will be added separately to keep the size of the PR manageable.

### Testing instructions
1. Prepare some test data: create some coupons, and add them to some orders (you can do it from wp-admin even for old orders).
2. Open the app.
3. Enable the coupons card.
4. Confirm the coupons are loaded for the selected range.
6. Change the range and confirm the coupons are fetched for the new one.
